File: Memento/cpp/Memento.cpp
#include <bits/stdc++.h>

using namespace std;

class Memento{
    private:
        string nombre;
        string apellidos;
    public:
        Memento(string,string);
        string getNombre();
        void setNombre(string);
        string getApellidos();
        void setApellidos(string);
};

Memento::Memento(string nombre, string apellidos){
    this->nombre=nombre;
    this->apellidos=apellidos;
}

string Memento::getNombre(){
    return this->nombre;
}
void Memento::setNombre(string nombre){
    this->nombre=nombre;
}
string Memento::getApellidos(){
    return this->apellidos;
}
void Memento::setApellidos(string apellidos){
    this->apellidos=apellidos;
}

File: Memento/Python/memento.py
class Memento:
    nombre=""
    apellidos=""

    def __init__(self,nombre="",apellidos=""):
        self.nombre=nombre
        self.apellidos=apellidos

class Originator:
    nombre=""
    apellidos=""

    def __init__(self,nombre,apellidos):
        self.nombre=nombre
        self.apellidos=apellidos
    
    def setMemento(self,m):
        self.nombre=m.nombre
        self.apellidos=m.apellidos

    def createMemento(self):
        return Memento(self.nombre,self.apellidos)

class Caretaker:
    memento=Memento()